src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -480,35 +183,66 @@ public async Task InitializeFromProfile(DacDeployOptions options, string profile
+            {
+                if (prop.PropertyType == typeof(System.Boolean))
+                {
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    this.BooleanOptionsDictionary[prop.Name] = (DeploymentOptionProperty<bool>)setProp;
+                }
+            }
+        }
+
         public void SetOptions(DacDeployOptions options)
         {
             System.Reflection.PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            // Set the default options properties
+            InitializeBooleanTypeOptions(options);
+            InitializeNonBooleanTypeOptions(options);
+        }
 
-            foreach (var deployOptionsProp in deploymentOptionsProperties)
+        /// <summary>
+        /// Preparing all non boolean properties (except BooleanOptionsDictionary)
+        /// </summary>
+        /// <param name="options"></param>
+        public void InitializeNonBooleanTypeOptions(DacDeployOptions options)
+        {
+            // preparing remaining properties
+            PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            foreach (PropertyInfo deployOptionsProp in deploymentOptionsProperties)
             {
-                var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-                // Note that we set excluded object types here since dacfx has this value as null;
-                if (prop != null)
+                if (deployOptionsProp.Name != nameof(DeploymentOptions.BooleanOptionsDictionary))
                 {
-                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
+                    var prop = options.GetType().GetProperty(deployOptionsProp.Name);
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    deployOptionsProp.SetValue(this, setProp);
                 }
             }
         }
 
         /// <summary>
-        /// Sets the Value and Description to all properties
+        /// Prepares and returns the value and description of a property
         /// </summary>
         /// <param name="prop"></param>
-        /// <param name="val"></param>
-        /// <param name="deployOptionsProp"></param>
-        public void SetGenericDeployOptionProps(PropertyInfo prop, DacDeployOptions options, PropertyInfo deployOptionsProp)
+        /// <param name="options"></param>
+        public object GetDeploymentOptionProp(PropertyInfo? prop, DacDeployOptions options)
         {
             var val = prop.GetValue(options);
-            var attribute = prop.GetCustomAttributes<DescriptionAttribute>(true).FirstOrDefault();
-            Type type = val != null ? typeof(DeploymentOptionProperty<>).MakeGenericType(val.GetType()) 
+            DescriptionAttribute descriptionAttribute = prop.GetCustomAttributes<DescriptionAttribute>(true).FirstOrDefault(); 
```

Comment:
FirstOrDefault can return null in these cases - make sure you're handling that appropriately (even if that's just throwing an exception with more detailed information instead of generic NullReferenceException)